v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandlerTest.java

```diff
@@ -0,0 +1,1318 @@
+    String colKey = "column";
+    IllegalArgumentException thrown =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              CassandraTypeHandler.handleStringifiedJsonToMap(colKey, newValuesJson);
+            });
+    assertTrue(thrown.getMessage().contains("Invalid stringified JSON format for column"));
+  }
+
+  @Test
+  public void testHandleStringifiedJsonToMap_NonJsonString() {
+    String newValuesString = "{\"column\": \"just a plain string\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "column";
+    IllegalArgumentException thrown =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              CassandraTypeHandler.handleStringifiedJsonToMap(colKey, newValuesJson);
+            });
+    assertTrue(thrown.getMessage().contains("Invalid stringified JSON format for column"));
+  }
+
+  @Test
+  public void testHandleCassandraVarintType_ValidByteArray() {
+    JSONObject valuesJson = new JSONObject();
+    byte[] byteArray = new BigInteger("12345678901234567890").toByteArray();
+    valuesJson.put("varint", byteArray);
+    BigInteger result = CassandraTypeHandler.handleCassandraVarintType("varint", valuesJson);
+    BigInteger expected = new BigInteger(byteArray);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void testHandleCassandraVarintType_InvalidStringFormat() {
+    JSONObject valuesJson = new JSONObject();
+    valuesJson.put("col1", "invalid-number");
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              handleCassandraVarintType("col1", valuesJson);
+            });
+    assertTrue(exception.getMessage().contains("Invalid varint format (string) for column: col1"));
+  }
+
+  @Test
+  public void testParseDate_UnsupportedType() {
+    JSONObject valuesJson = new JSONObject();
+    valuesJson.put("col1", 12345);
+    String formatter = "yyyy-MM-dd";
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              CassandraTypeHandler.parseDate("col1", valuesJson.get("col1"), formatter);
+            });
+    assertTrue(exception.getMessage().contains("Unsupported type for column col1"));
+  }
+
+  @Test
+  public void testHandleCassandraUuidType_ValidUuidString() {
+    JSONObject valuesJson = new JSONObject();
+    String validUuidString = "123e4567-e89b-12d3-a456-426614174000";
+    valuesJson.put("col1", validUuidString);
+    UUID result = handleCassandraUuidType("col1", valuesJson);
+    UUID expectedUuid = UUID.fromString(validUuidString);
+    assertEquals(expectedUuid, result);
+  }
+
+  @Test
+  public void testHandleCassandraInetAddressType_Hostname() {
+    JSONObject valuesJson = new JSONObject();
+    valuesJson.put("col1", "www.google.com");
+    IllegalArgumentException exception =
```

Comment:
Thanks for adding this test case.

[super-nit]
Could we consistenty use `assertThrows` for the unit tests which need asserts?

`assertThrows` is the strongly preferred idiom. [go/java-testing/exceptions#assertThrows](http://go/java-testing/exceptions#assertThrows) And
[exceptions#discouraged](http://go/java-testing/exceptions#discouraged)

Reply:
Sure we will make this consistent change in the following PRs